sabnzbd/getipaddress.py

```diff
@@ -101,6 +101,7 @@ def publicipv4():
             # specify the Host, because we only provide the IPv4 address in the URL:
             req.add_header("Host", sabnzbd.cfg.selftest_host())
             # get the response, timeout 2 seconds, in case the website is not accessible
+            socket.socket = sabnzbd.ORIGINAL_SOCKET
```

Comment:
I don't understand why we would want to return the original socket? When using a proxy wouldn't you want to see exactly the result _with_ proxy?
Would also simplify even some more code so we don't have to switch sockets.
We can just execute it once at startup and done!

Reply:
I was assuming it's trying to find the public IP of the server so that other servers can contact it. That won't work if it uses the proxy. In general, removing the proxy server settings will require a restart if the socket is not stored. As far as I have been able to find out it's not possible to remove the proxy settings from the socket.

Reply:
I could add a callback to `set_socks5_proxy()` so that it's updated after editing instead of setting it before every request.

Regarding `publicipv4()` I think it should either return the IP of the user or nothing if it's using a proxy, not the IP of the proxy. I can't think of any use cases where that would be useful. In this case I suggest setting the original socket before the request and calling `set_socks5_proxy()` after.

Reply:
I am not a fan of these special set and unset steps, especially for exotic features like these.

Reply:
Especially so we don't have to think about it and possibly forget some request.
So just calling it once at startup should be enough and we can forget about it :)